src/lib/commandclass/ColorSwitchCC.ts

```diff
@@ -0,0 +1,762 @@
+				label: `Color ${colorName}`,
+				description: `The target level of the ${colorName} color.`,
+				min: 0,
+				max: 255,
+			});
+			node.valueDB.setValue(targetValueId, this._targetValue);
+		}
+	}
+
+	private _colorComponent: ColorComponent;
+	public get colorComponent(): ColorComponent {
+		return this._colorComponent;
+	}
+
+	private _currentValue: number;
+	public get currentValue(): number {
+		return this._currentValue;
+	}
+
+	private _targetValue: number | undefined;
+	public get targetValue(): number | undefined {
+		return this._targetValue;
+	}
+
+	private _duration: Duration | undefined;
+	public get duration(): Duration | undefined {
+		return this._duration;
+	}
+}
+
+interface ColorSwitchCCGetOptions extends CCCommandOptions {
+	colorComponent: ColorComponent;
+}
+
+const testResponseForColorSwitchGet: CCResponsePredicate = (
+	sent: ColorSwitchCCGet,
+	received,
+	isPositiveTransmitReport,
+) => {
+	return received instanceof ColorSwitchCCReport &&
+		sent.colorComponent === received.colorComponent
+		? "final"
+		: // TODO: What is isPositiveTransmitReport?  Do I need it?
+		isPositiveTransmitReport
+		? "confirmation"
+		: "unexpected";
+};
+
+@CCCommand(ColorSwitchCommand.Get)
+@expectedCCResponse(testResponseForColorSwitchGet)
+export class ColorSwitchCCGet extends ColorSwitchCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions | ColorSwitchCCGetOptions,
+	) {
+		super(driver, options);
+		if (gotDeserializationOptions(options)) {
+			// TODO: Deserialize payload
+			throw new ZWaveError(
+				`${this.constructor.name}: deserialization not implemented`,
+				ZWaveErrorCodes.Deserialization_NotImplemented,
+			);
+		} else {
+			// Populate properties from options object
+			this._colorComponent = options.colorComponent;
+		}
+	}
+
+	private _colorComponent: ColorComponent;
+	public get colorComponent(): ColorComponent {
+		return this._colorComponent;
+	}
+	public set colorComponent(value: ColorComponent) {
+		if (!ColorComponent[value]) {
+			throw new Error(
+				"colorComponent must be a valid color component index.",
+			);
+		}
+		this._colorComponent = value;
+	}
+
+	public serialize(): Buffer {
+		this.payload = Buffer.allocUnsafe(1);
+		this.payload[0] = this._colorComponent;
+		return super.serialize();
+	}
+}
+
+export interface ColorSwitchCCSetOptions extends CCCommandOptions, ColorTable {
+	duration?: Duration;
+}
+
+@CCCommand(ColorSwitchCommand.Set)
+export class ColorSwitchCCSet extends ColorSwitchCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions | ColorSwitchCCSetOptions,
+	) {
+		super(driver, options);
+		if (gotDeserializationOptions(options)) {
+			// TODO: Deserialize payload
+			throw new ZWaveError(
+				`${this.constructor.name}: deserialization not implemented`,
+				ZWaveErrorCodes.Deserialization_NotImplemented,
+			);
+		} else {
+			// Populate properties from options object
+			this._colorTable = {
```

Comment:
I'm doing this so often, I should probably add a `pick` utility method to pick properties from an object.

Reply:
I added a pick() function in util/misc and made use of it here.